Update Jenkinsfile to use weekly CI results

yiqian1 opened the pull request:
- Nightly CI copies the achieved user perfDb from the most recent successful weekly for performance tests.
- Nightly CI runs MIOpen tests on two selected Resnet50 configs (filter 1x1 and 3x3).

These changes have passed validations:
http://ml-ci.amd.com:21096/job/MLIR/job/test-nightlyCI/13/
http://rocmhead:8080/job/test-nightlyCI/3/

Weekly CI job's name is mlir-weekly.
http://ml-ci.amd.com:21096/job/MLIR/job/mlir-weekly/2/
http://rocmhead:8080/job/mlir-weekly/4/

yiqian1 committed:
Update Jenkinsfile to use weekly CI results

jerryyin commented on mlir/utils/jenkins/Jenkinsfile:
```diff
@@ -410,14 +419,14 @@ pipeline {
+                        copyArtifacts filter: 'MIOpen/build/MIOpenUserDB/**',\
+                                               optional: true,\
+                                               flatten: true,\
+                                               projectName: "mlir-weekly",\
```
Just curious: You already have the weekly job setup? Mind amend the confluence page to remind the new job?

yiqian1 replied:
Yes. I have to. The weekly results are needed for nightlies.
 Will update the confluence page.

jerryyin commented on mlir/utils/jenkins/Jenkinsfile:
```diff
@@ -264,18 +264,6 @@ pipeline {
                                 }
                             }
                         }
-                        stage("Tune MLIR Kernels") {
-                            when { allOf{
-                                equals expected: true, actual: params.nightly;
-                                equals expected: true, actual: params.perfTest;
-                            } }
-                            steps {
-                                buildProject('ci-performance-scripts', '')
-                                dir('build') {
-                                    sh 'python3 ./bin/MIOpenDriver.py -tuning'
-                                }
-                            }
-                        }
```
Does this still need to exist under weekly context? Or else which stage do the tuning now?

yiqian1 replied:
It is now done in the "Test MIOpen config with tuning" stage in weeklies.

jerryyin replied:
Could you amend that stage name to this one:

Now that stage isn't a test any more but more precisely the `Tune MLIR Kernels` work.

jerryyin commented on mlir/utils/jenkins/Jenkinsfile:
```diff
@@ -304,18 +292,39 @@ pipeline {
                             }
                         }
- 
+                        stage("Test selected MIOpen configs") {
```
How long does this stage take and is there any possibility to make it part of PR CI if it is short enough?

jerryyin replied:
To rephrase, I'm also curious how long does it take for all configs to run through tuning/non-tuning? If it is 2 mins perf flavor, we have a total of 24 flavors than in total it should consume around 50 mins. It sounds do-able as part of nightly as well.

yiqian1 replied:
> How long does this stage take and is there any possibility to make it part of PR CI if it is short enough?

20-30 minutes. 
BTW, tuning on the two selected configs takes 32 minutes and non-tuning takes 1 minutes on private CI. Plus 10 minutes for building MIOpenDriver.  Running this stage for PR may increase the time of PR CI by 45 minutes.

yiqian1 replied:
> To rephrase, I'm also curious how long does it take for all configs to run through tuning/non-tuning? If it is 2 mins perf flavor, we have a total of 24 flavors than in total it should consume around 50 mins. It sounds do-able as part of nightly as well.

Tuning takes very long time to finish. 3 hours on public CI and 6 hours on private CI. non-tuning takes about 30 minutes to finish. In theory it can take as little at 10 minutes if there are enough CI nodes available.

jerryyin replied:
20 - 30 mins does seem to be reasonable to add to PR CI, I'd prefer to do that so we have minimal coverage on PR submissions.

On the other hand, I wasn't talking about the tuning itself, but to test the tuned configs based on weekly result. That should be roughly the same amount of test on testing without perfdb.

jerryyin replied:
So just to be clear I'm proposing this:
 - PR test: test selected MIOpen configs
 - nightly test: test `MIOpen Resnet50 config Test`
 - Weekly test stay the same and do the actual tuning

yiqian1 replied:
> 20 - 30 mins does seem to be reasonable to add to PR CI, I'd prefer to do that so we have minimal coverage on PR submissions.
> 
> On the other hand, I wasn't talking about the tuning itself, but to test the tuned configs based on weekly result. That should be roughly the same amount of test on testing without perfdb.

The performance tests are run with and without perfDb (only for NCHW). Each run (tuning or non-tuning) takes about 10 minutes, or the total of 20 minutes. 

Should we also do it for NHWC?

yiqian1 replied:
> So just to be clear I'm proposing this:
> 
> * PR test: test selected MIOpen configs
> * nightly test: test `MIOpen Resnet50 config Test`
> * Weekly test stay the same and do the actual tuning

OK.

yiqian1 replied:
Is it good enough if we do selected MIOpen config tests only on private PR CI? 

Also I want to verify that:

- PR CI: selected MIOpen configs tests including tuning and tests without perfDb. 
- nightly: MIOpen Resnet50 config Test including tests with and without perfDb.
- weekly: Tune the MLIR kernels. No need to do "MIOpen Resnet50 config Test"

Am I right?

jerryyin commented on mlir/utils/jenkins/miopen-tests/selected-resnet50-miopen-configs:
```diff
@@ -0,0 +1,2 @@
+-n 256 -c 1024 -H 14 -W 14 -k 256 -y 1 -x 1 -p 0 -q 0 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -t 1
```
Irrelevant to this file: Why the rename from resnext101 -> resnet101? I think you should revert the rename.

yiqian1 committed:
Rearrange Jenkins tasks

-Perform selected MIOpen tests in PR CI
-Perform full MIOpen tests with and without perfDb in nightly CI
-Remove MIOpen tests from weekly CI

yiqian1 committed:
Perform selected MIOpen tests in public PR CI

krzysz00 commented on mlir/utils/jenkins/Jenkinsfile:
```diff
@@ -508,6 +509,14 @@ pipeline {
+                                    copyArtifacts filter: 'MIOpen/build/MIOpenUserDB/**',\
+                                               optional: true,\
+                                               flatten: true,\
+                                               projectName: "mlir-weekly",\
```
Since this bit me before - on public CI, the project might be named MLIR/mlir-weekly and IIRC the perf tests have some workaround for that.

There's also permissions issues that can come up with artifacts. Let's make sure this works before merge.

yiqian1 replied:
The project name uses a relative path here. It works for both CIs.

krzysz00 reviewed (commented):
I don't see too many issues with the PR as is, other than that it'll need testing to make sure the artifact publishing and pulling works on both CIs

jerryyin commented on mlir/utils/jenkins/Jenkinsfile.downstream:
```diff
@@ -7,6 +7,34 @@ void buildProject(String target, String cmakeOpts) {
         cmakeArgs: "-DMLIR_MIOPEN_DRIVER_ENABLED=1 $cmakeOpts"
 }
 
+void buildMIOpen(String cmakeOpts) {
```
Not an issue in this PR but we may start to consider to let two jenkinsfile reuse those common utilities in a third file.

jerryyin reviewed (approved):
LGTM, nothing past the existing comments: Please still finish the confluence page and send a email to the team to let them know about the weekly CI.
